lib/mogrify.ex

```diff
@@ -37,11 +58,23 @@ defmodule Mogrify do
     end
   end
 
-  defp arguments(image, path) do
+  defp arguments_for_saving(image, path) do
     base_arguments = ~w(-write #{path} #{String.replace(image.path, " ", "\\ ")})
-    additional_arguments = Enum.flat_map image.operations, fn {option,params} -> ~w(-#{option} #{params}) end
+    arguments(image) ++ base_arguments
+  end
+
+  defp arguments_for_creating(image, path) do
+    base_arguments = ~w(#{Path.dirname(path)}/#{String.replace(Path.basename(image.path), " ", "\\ ")})
+    arguments(image) ++ base_arguments
+  end
 
-    additional_arguments ++ base_arguments
+  defp arguments(image) do
+    Enum.flat_map(image.operations, fn {option,params} ->
+      case option do
```

Comment:
I'm not a big fan of the extra closure, it's pretty hard to read, imo. I'm ok with a `defp`.